tests/async_client_tests.rs

```diff
@@ -53,9 +51,42 @@ fn sync() {
 
     let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
 
-    let response = rt.block_on(client.sync(sync_settings)).unwrap();
+    let response = client.sync(sync_settings).await.unwrap();
 
     assert_ne!(response.next_batch, "");
 
-    assert!(rt.block_on(client.sync_token()).is_some());
+    assert!(client.sync_token().await.is_some());
+}
+
+#[tokio::test]
```

Comment:
Could you add some more tests? This already adds quite a bit functionality and we're not covering much.

E.g. room display name calculation isn't tested at all. We track coverage with [tarpaulin](https://github.com/xd009642/tarpaulin), it just isn't yet set up for CI.

Reply:
I'm having troubles installing tarpaulin but I think I may have just figured it out.

Do you want me to add internal tests for the `pub(crate)` methods, for `Room`, `RoomMember`, the base client, ect.

Reply:
If you have the time, sure. The more tests the better.

Reply:
I got `tarpaulin` to install yay! haha stupid `libssh`